src/control/cmd/daos_agent/mgmt_rpc.go

```diff
@@ -130,14 +129,18 @@ func (mod *mgmtModule) handleGetAttachInfo(ctx context.Context, reqb []byte, pid
 	return proto.Marshal(resp)
 }
 
+func providerIsVerbs(provider string) bool {
+	return strings.HasPrefix(provider, verbsProvider)
+}
+
 func (mod *mgmtModule) getAttachInfo(ctx context.Context, numaNode int, sys string) (*mgmtpb.GetAttachInfoResp, error) {
 	resp, err := mod.getAttachInfoResp(ctx, numaNode, sys)
 	if err != nil {
 		mod.log.Errorf("failed to fetch remote AttachInfo: %s", err.Error())
 		return nil, err
 	}
 
-	fabricIF, err := mod.getFabricInterface(ctx, numaNode, resp.ClientNetHint.NetDevClass)
+	fabricIF, err := mod.getFabricInterface(ctx, numaNode, resp.ClientNetHint.NetDevClass, providerIsVerbs(resp.ClientNetHint.Provider))
```

Comment:
I think I've finally put my finger on what makes me uneasy about this PR... Instead of making the decision out here about whether or not the domain is expected, wouldn't it be better to keep the business logic for handling domains more contained in the infocache/fabric stuff? In other words, instead of adding a boolean to this method, instead just pass in the provider from the hint. This would also help to make the code a bit more future-proof, I think.